From: Paul Boddie <paul@boddie.org.uk>
Date: Tue, 15 Aug 2023 23:50:33 +0200
Subject: Fix package data installation.

The provided package_data settings for static data and themes are
insufficient and will not copy the data. A function has been introduced
to identify the data files in each location.

moin (2.0+git20230330.c55318e3-1) unstable; urgency=medium

  * Initial release.
---
 setup.py | 26 +++++++++++++++++---------
 1 file changed, 17 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index 93fcf7e..59de323 100644
--- a/setup.py
+++ b/setup.py
@@ -6,6 +6,7 @@
 
 import os
 from setuptools import setup, find_packages
+from glob import glob
 
 
 basedir = os.path.abspath(os.path.dirname(__file__))
@@ -13,6 +14,18 @@ with open(os.path.join(basedir, 'README.rst')) as f:
     long_description = f.read()
 
 
+def find_files(base, names):
+    base = os.path.join(basedir, base)
+    base = os.path.join(os.path.normpath(base), "")
+    l = []
+    for name in names:
+        d = os.path.join(base, name)
+        pattern = os.path.join(d, "**")
+        for s in glob(pattern, recursive=True):
+            l.append(s[len(base):])
+    return l
+
+
 setup_args = dict(
     name="moin",
     python_requires='>=3.8',
@@ -53,15 +66,10 @@ setup_args = dict(
     package_dir={'': 'src'},
 
     package_data={
-        'moin.translations': ['MoinMoin.pot', '*.po', ],
-        'moin.static': ['*', ],
-        'moin.themes.modernized': ['*', ],
-        'moin.themes.basic': ['*', ],
-        'moin.themes.topside': ['*', ],
-        'moin.themes.topside_cms': ['*', ],
-        'moin.templates': ['*.html', '*.xml', ],
-        'moin.apps.admin.templates': ['*.html', ],
-        'moin.apps.misc.templates': ['*.html', '*.txt', ],
+        'moin': find_files('src/moin', ['static', 'templates', 'translations']),
+        'moin.themes': find_files('src/moin/themes', ['modernized', 'basic', 'topside', 'topside_cms']),
+        'moin.apps.admin': find_files('src/moin/apps/admin', ['templates']),
+        'moin.apps.misc': find_files('src/moin/apps/misc', ['templates']),
         'moin.contrib': ['*.txt', ],
     },
     include_package_data=True,
